claripy/frontends/portfolio_frontend.py

```diff
@@ -0,0 +1,254 @@
+import logging
+from multiprocessing import Process, Pool
+from functools import partial
+
+from .constrained_frontend import ConstrainedFrontend
+from ..frontend import Frontend
+
+import traceback
+
+l = logging.getLogger("claripy.frontends.portfolio_frontend")
+
+
+def execute_solver_satisfiable(args):
+    solver, extra_constraints, exact = args
+    try:
+        satness = solver.check_satisfiability(
+            extra_constraints=extra_constraints,
+            exact=exact
+        )
+        if satness == 'UNKNOWN':
+            return solver, None
+        return solver, satness == 'SAT'
+    except Exception as e:
+        traceback.print_exc()
+        l.error("Solver failed to check satisfiability because %r", e)
+        return solver, None
+
+
+def execute_solver_eval(args):
+    solver, e, n, extra_constraints, exact = args
+    try:
+        res = solver.eval(
+            e, n, extra_constraints=extra_constraints, exact=exact
+        )
+        return solver, res
+    except Exception as e:
+        traceback.print_exc()
+        l.error("Solver failed to get solution because %r", e)
+        return solver, None
+
+
+class PortfolioFrontend(Frontend):
+
+    def __init__(self, solvers, *args, **kwargs):
+        super(PortfolioFrontend, self).__init__()
+        self.solvers = solvers
+
+    # def _blank_copy(self, c):
+    #     super(PortfolioFrontend, self)._blank_copy(c)
+    #     c.solvers = self.solvers
+    #
+    # def _copy(self, c):
+    #     super(PortfolioFrontend, self)._copy(c)
+    #     c.solvers = list(self.solvers)
+
+    def branch(self):
+        c = self.blank_copy()
+        c.solvers = [solver.branch() for solver in self.solvers]
+        return c
+
+    #
+    # #
+    # # Storable support
+    # #
+    #
+    # def _ana_getstate(self):
+    #     return self._solver_backend.__class__.__name__, self.timeout, self._track, ConstrainedFrontend._ana_getstate(self)
+    #
+    # def _ana_setstate(self, s):
+    #     backend_name, self.timeout, self._track, base_state = s
+    #     self._solver_backend = backends._backends_by_type[backend_name]
+    #     #self._tls = None
+    #     self._tls = threading.local()
+    #     self._to_add = [ ]
+    #     ConstrainedFrontend._ana_setstate(self, base_state)
+    #
```

Comment:
`ana` is gone for good. Let's just remove those lines.

Reply:
Fixed now